src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -187,7 +205,7 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 		this._processStartupComplete = new Promise<void>(c => {
 			this.onProcessReady(() => c());
 		});
-		ptyProcess.onData(data => {
+		ptyProcess.onData(async data => {
```

Comment:
While check shell is async, we don't right now and there's no need to await it so we can use a regular callback:

```suggestion
		ptyProcess.onData(data => {
```